web/src/enterprise/campaigns/detail/CampaignStatus.tsx

```diff
@@ -1,102 +1,118 @@
-import CheckCircleIcon from 'mdi-react/CheckCircleIcon'
-import AlertCircleIcon from 'mdi-react/AlertCircleIcon'
-import { ErrorAlert } from '../../../components/alerts'
-import InformationIcon from 'mdi-react/InformationIcon'
-import { parseISO, isBefore, addMinutes } from 'date-fns'
+import { ErrorMessage } from '../../../components/alerts'
+import SyncIcon from 'mdi-react/SyncIcon'
+import { pluralize } from '../../../../../shared/src/util/strings'
 
 export interface CampaignStatusProps {
     campaign: Pick<GQL.ICampaign, 'closedAt' | 'viewerCanAdminister' | 'publishedAt'> & {
         changesets: Pick<GQL.ICampaign['changesets'], 'totalCount'>
+        status: Pick<GQL.ICampaign['status'], 'completedCount' | 'pendingCount' | 'errors' | 'state'>
     }
 
-    /** The campaign status. */
-    status: Omit<GQL.IBackgroundProcessStatus, '__typename'>
-
     /** Called when the "Publish campaign" button is clicked. */
     onPublish: () => void
     /** Called when the "Retry failed jobs" button is clicked. */
     onRetry: () => void
 }
 
+type CampaignState = 'closed' | 'errored' | 'processing' | 'completed'
+
 /**
  * The status of a campaign's jobs, plus its closed state and errors.
  */
-export const CampaignStatus: React.FunctionComponent<CampaignStatusProps> = ({
-    campaign,
-    status,
-    onPublish,
-    onRetry,
-}) => {
-    /* For completed campaigns that have been published, hide the creation complete status 1 day after the time of publication */
-    const creationCompletedLongAgo =
-        status.state === GQL.BackgroundProcessState.COMPLETED &&
-        !!campaign.publishedAt &&
-        isBefore(parseISO(campaign.publishedAt), addMinutes(new Date(), 1))
+export const CampaignStatus: React.FunctionComponent<CampaignStatusProps> = ({ campaign, onPublish, onRetry }) => {
+    const { status } = campaign
+
     const progress = (status.completedCount / (status.pendingCount + status.completedCount)) * 100
-    return (
-        <>
-            {status.state === GQL.BackgroundProcessState.PROCESSING && (
-                <div className="mt-3 e2e-preview-loading">
-                    <div className="progress mb-1">
-                        {/* we need to set the width to control the progress bar, so: */}
-                        {/* eslint-disable-next-line react/forbid-dom-props */}
-                        <div className="progress-bar" style={{ width: progress + '%' }}>
-                            &nbsp;
-                        </div>
+
+    const isDraft = !campaign.publishedAt
+    let state: CampaignState
+    if (campaign.closedAt) {
+        state = 'closed'
+    } else if (campaign.status.state === GQL.BackgroundProcessState.ERRORED) {
+        state = 'errored'
+    } else if (campaign.status.state === GQL.BackgroundProcessState.PROCESSING) {
+        state = 'processing'
+    } else {
+        state = 'completed'
+    }
+
+    let statusIndicatorComponent: JSX.Element | undefined
```

Comment:
Nit: this is an element, not a component - a component is a function that returns elements
```suggestion
    let statusIndicator: JSX.Element | undefined
```